From 3536ef6bb74144cf9dce10200cd62c2ad9f7cae4 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Thu, 15 Jan 2026 19:36:17 -0800
Subject: [PATCH] tools/elfdeps.cc: Remove format module

There is no format until gcc 13.1, so remove the format module to make it work
on hosts such as Ubuntu 22.04.

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
Signed-off-by: Sumanth Gavini <sumanth.gavini@windriver.com>
---
 tools/elfdeps.cc | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/tools/elfdeps.cc b/tools/elfdeps.cc
index 17884b24e..33ff8d740 100644
--- a/tools/elfdeps.cc
+++ b/tools/elfdeps.cc
@@ -1,6 +1,5 @@
 #include "system.h"
 
-#include <format>
 #include <string>
 #include <vector>
 
@@ -105,7 +104,7 @@ static void addSoDep(std::vector<std::string> & deps,
     if (ver.empty() && marker.empty()) {
 	addDep(deps, soname);
     } else {
-	auto dep = std::format("{}({}){}", soname, ver, marker);
+	std::string dep = soname + "(" + ver + ")" + marker;
 	addDep(deps, dep);
     }
 }
-- 
2.49.0

